Formula/d/damask-grid.rb

```diff
@@ -0,0 +1,54 @@
+class DamaskGrid < Formula
+  desc "Grid solver of the Multi-physics simulation package - DAMASK"
+  homepage "https://damask-multiphysics.org"
+  url "https://damask-multiphysics.org/download/damask-3.0.0.tar.xz"
+  sha256 "aaebc65b3b10e6c313132ee97cfed427c115079b7e438cc0727c5207e159019f"
+  license "AGPL-3.0-only"
+
+  depends_on "cmake" => :build
+  depends_on "pkg-config" => :build
+  depends_on "fftw"
+  depends_on "hdf5-mpi"
+  depends_on "petsc"
+
+  uses_from_macos "zlib"
+
+  patch :p1, :DATA
+
+  def install
+    ENV["PETSC_DIR"] = Formula["petsc"].opt_prefix
+    args = %w[
+      -DDAMASK_SOLVER=grid
+      -DCMAKE_INSTALL_PREFIX=${PWD}
+    ]
+    system "cmake", "-S", ".", "-B", "build-grid", *args, *std_cmake_args
+    system "cmake", "--build", "build-grid", "--target", "install"
+  end
+
+  test do
+    shell_output("$(brew --prefix damask-grid)/bin/DAMASK_grid")
```

Comment:
Is there some output we can check here?

```suggestion
    assert_match "some output", shell_output("DAMASK_grid")
```